File: Server/DataBase/Entities/RecipeStepEntity.cs
namespace Server.DataBase.Entities;

public class RecipeStepEntity
{
    public int Id { get; init; }
    public int RecipeEntityId { get; init; }
    public int StepNumber { get; set; }
    public string Description { get; set; } = null!;
    public string ImageName { get; set; } = null!;
    public RecipeEntity RecipeEntity { get; set; }
}

File: client/src/models/RecipeStepEntity.js
class RecipeStepEntity {
    constructor(json) {
        if (json) {
            this.stepNumber = json.stepNumber;
            this.description = json.description;
            this.imageName = json.imageName;
        } else {
            this.stepNumber = 1;
            this.description = '';
            this.imageName = '';
        }
    }
}

export default RecipeStepEntity;